### PR TITLE
V1.5.1-beta

### PR DESCRIPTION
**Pull request by arindam1993**
Adds the following fixes to release-sangria
- Fix animation performance: https://github.com/mapbox/mapbox-gl-js/pull/8913
- Fix memory leak with Cache API in Safari https://github.com/mapbox/mapbox-gl-js/pull/8956
- Workaround memory leak caused by using transferable objects https://github.com/mapbox/mapbox-gl-js/pull/9003

### CODE REVIEW

File: src/util/web_worker_transfer.js [hunk 1]
```diff
@@ -119,7 +119,7 @@ function isArrayBuffer(val: any): boolean {
  *
  * @private
  */
-export function serialize(input: mixed, transferables?: Array<Transferable>): Serialized {
+export function serialize(input: mixed, transferables: ?Array<Transferable>): Serialized {
```
Comment (ahk) [inline src/util/web_worker_transfer.js]: I'm surprised these aren't equivalent. What's the difference? An optional parameter vs a parameter with an optional value ...

File: src/util/util.js [hunk 2]
```diff
@@ -409,6 +409,18 @@ export function sphericalToCartesian([r, azimuthal, polar]: [number, number, num
     };
 }
 
+/* global self, WorkerGlobalScope */
+/**
+ *  Retuns true if the when run in the web-worker context.
```
Comment (ahk) [inline src/util/util.js]: Typo of 'Returns'

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,11 @@
...
+
+## 🍏 Improvements
+* Implement workaround for memory leak in Safari when using the Cache API. ( [#8856](https://github.com/mapbox/mapbox-gl-js/pull/8956))
+* Implement workaround for memory leak in Safari when using Transferable objects to transfer ArrayBuffers. This involves browser detecting Safari and disabling the use of Transferables when posting messags to WebWorkers. ( [#9003](https://github.com/mapbox/mapbox-gl-js/pull/9003))
```
[the remark this answers is not in the document]
Comment (chloekraw) [reply CHANGELOG.md]: nit: I feel the second sentence could be cleaned up. how about something like:

> ...`ArrayBuffers`. If the Safari browser is detected, this workaround disables the use of `Transferables` when posting messages (note the typo) to `WebWorkers`.

File: src/util/web_worker_transfer.js (hunk 1 above)
Comment (kkaefer) [reply src/util/web_worker_transfer.js]: [An optional parameter accepts `undefined | T`, while an optional type accepts  `undefined | null | T`.](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVUCuA7AxgFwEs5swo44AKADwC4wB+bTAWwCMBTAJwEp6A3OIQAmYAN6oAvunJUeAbgwVKzGDAVKqOYRyiFsHYRtmUATBow4CxUmwCGXGg3rN23PmEEjxU9Pcca-iqYaoEOlNq6+oZhjubyQA)

**Pull request merged**